corehq/apps/app_manager/suite_xml/sections/menus.py

```diff
@@ -1,15 +1,22 @@
 from corehq.apps.app_manager import id_strings
-from corehq.apps.app_manager.exceptions import ScheduleError
+from corehq.apps.app_manager.exceptions import ScheduleError, CaseXPathValidationError
 from corehq.apps.app_manager.suite_xml.contributors import SuiteContributorByModule
 from corehq.apps.app_manager.suite_xml.xml_models import Menu, Command, LocalizedMenu
-from corehq.apps.app_manager.util import is_usercase_in_use
-from corehq.apps.app_manager.xpath import interpolate_xpath, CaseIDXPath, session_var, QualifiedScheduleFormXPath
+from corehq.apps.app_manager.util import is_usercase_in_use, xpath_references_case
+from corehq.apps.app_manager.xpath import (interpolate_xpath, CaseIDXPath, session_var,
+    QualifiedScheduleFormXPath, CASE_REFERENCE_VALIDATION_ERROR)
 from corehq.feature_previews import MODULE_FILTER
 
 
+def disallow_xpath_case_references(xpath):
+    if xpath_references_case(xpath):
+        raise CaseXPathValidationError(CASE_REFERENCE_VALIDATION_ERROR)
+
+
 class MenuContributor(SuiteContributorByModule):
     def get_module_contributions(self, module):
         def get_commands():
+            module_uses_case = module.all_forms_require_a_case() or is_usercase_in_use(self.app.domain)
```

Comment:
Could we make this lazy to avoid the DB hit in `is_usercase_in_use` unless it's needed? Also this is what's causing the tests to fail in a non-db test case.